web/src/compose_closed_ui.ts

```diff
@@ -171,24 +171,21 @@ export function set_standard_text_for_reply_button(): void {
 export function update_reply_recipient_label(message?: ComposeClosedMessage): void {
     const recipient_label = get_recipient_label(message);
     if (recipient_label !== undefined) {
-        if (!recipient_label.has_empty_string_topic) {
-            const recipient_label_text = recipient_label.label_text;
-            set_reply_button_label(
-                $t({defaultMessage: "Message {recipient_label_text}"}, {recipient_label_text}),
-            );
-        } else {
-            const topic_display_name = util.get_final_topic_display_name("");
-            const recipient_label_html = $t_html(
-                {
-                    defaultMessage: "Message <z-recipient-label></z-recipient-label>",
-                },
-                {
-                    "z-recipient-label": () =>
-                        `#${recipient_label.stream_name} > <span class="empty-topic-display">${topic_display_name}</span>`,
+        const recipient_label_html = $t_html(
+            {
+                defaultMessage: "Message <z-recipient-label></z-recipient-label>",
+            },
+            {
+                "z-recipient-label"() {
+                    if (recipient_label.has_empty_string_topic) {
+                        const topic_display_name = util.get_final_topic_display_name("");
+                        return `#${recipient_label.stream_name} > <span class="empty-topic-display">${topic_display_name}</span>`;
+                    }
+                    return recipient_label.label_text;
                 },
-            );
-            $("#left_bar_compose_reply_button_big").html(recipient_label_html);
-        }
+            },
+        );
+        $("#left_bar_compose_reply_button_big").html(recipient_label_html);
```

Comment:
I'm concerned that this implementation has XSS issues. I think you should probably rework this into a Handlebars template the HTML generation here; I think you'll find that makes it a lot easier to be explicit about the distinction between user data and HTML.

Reply:
reworked this with a separate handlebar template approach.